HACKING.md

```diff
@@ -63,17 +63,23 @@ Sometimes however, updates to minikube may prevent the test cluster context bein
                                                                                             
 ## Make targets
 
-### Building Docker images
+Strimzi includes a `Makefile` with various Make targets to build the project.
 
-The `docker_build` target will build the Docker images provided by the Strimzi project. You can build all Strimzi Docker images by calling `make docker_build` from the root of the Strimzi repository. Or you can build an individual Docker image by running `make docker_build` from the subdirectories with their respective Dockerfiles - e.g. `kafka_base`, `kafka` etc.
+Commonly used Make targets:
 
-The `docker_build` target will **always** build the images under the `strimzi` organization. This is necessary in order to be able to reuse the base image you might have just built without modifying all Dockerfiles. The `DOCKER_TAG` environment variable configures the Docker tag to use (default is `latest`).
+ - `docker_build` for [building Docker images](#building-docker-images)
+ - `docker_tag` for [tagging existing images](#tagging-and-pushing-docker-images)
+ - `docker_push` for [pushing images to a Docker registry](#tagging-and-pushing-docker-images)
```

Comment:
@ppatierno it did feel a bit strange referring to the same paragraph with two different commands.. this was my thought process.. I have been particularly using the option of building using the daemon in Minikube, so I have been using `docker_tag` separately as I didn't want `docker_push` to happen, so it felt right to call it out as a useful task. But then I couldn't see an obvious way to break up the paragraph without having repetition between them.

Reply:
I see your point, so I think it's a little bit misleading the paragraph we already have in place ...

> Target `docker_tag` can be used to tag the Docker images built by the `docker_build` target. This target is automatically called by the `docker_push` target and doesn't have to be called separately.

of course you can use `docker_tag` without doing a push while reading the above paragraph, the "have to be called" part seems to mean that you can't ... at least it's how it sounds to me (as non native English speaker). @tomncooper wdyt?

Reply:
How about something like: 

> Target `docker_tag` tags the Docker images built by the `docker_build` target. This target is automatically called as part of the `docker_push` target, but can be called separately if you wish to avoid pushing images to an external registry.

Reply:
yeah it sounds better! @katheris wdyt? could you change in this way in your PR?

Reply:
@ppatierno @tomncooper sounds good, I'll make that update now

Reply:
Updated